src/App.tsx

```diff
@@ -1,26 +1,135 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useContext, useRef, useEffect,
+} from 'react';
+import classNames from 'classnames';
+import { TodoContext } from './contexts/TodoContext';
+import { ErrorType } from './types/ErrorType';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const {
+    todos,
+    tempTodo,
+    changeData,
+    dataError,
+    setError,
+    Error,
+    addTodo,
+    inputValue,
+    setInputValue,
+    shouldFocus,
+  } = useContext(TodoContext);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  let toggleActive = todos.every(currentTodo => currentTodo.completed);
+
+  useEffect(() => {
+    if (shouldFocus) {
+      inputRef.current?.focus();
+    } else {
+      inputRef.current?.blur();
+    }
+  }, [dataError, todos]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {todos.length !== 0 && (
+            <button
+              type="button"
+              data-cy="ToggleAllButton"
+              className={classNames(
+                'todoapp__toggle-all',
+                { active: todos.every(currentTodo => currentTodo.completed) },
+              )}
+              onClick={() => {
+                toggleActive = !toggleActive;
+
+                if (toggleActive) {
+                  todos.forEach(currentTodo => {
+                    if (currentTodo.completed === false) {
+                      changeData(
+                        currentTodo.id,
+                        currentTodo.title,
+                        true,
+                      );
+                    }
+                  });
+                }
+
+                if (!toggleActive) {
+                  todos.forEach(currentTodo => {
+                    if (currentTodo.completed === true) {
+                      changeData(
+                        currentTodo.id,
+                        currentTodo.title,
+                        false,
+                      );
+                    }
+                  });
+                }
+              }}
+            />
```

Comment:
Move this logic out of jsx? create separate handler and pass reference onClick prop